lib/logster/logger.rb

```diff
@@ -74,9 +74,13 @@ def add_with_opts(severity, message = nil, progname = progname(), opts = nil, &b
       # own methods don't show up as the first few frames in the backtrace
       if !opts || !opts.key?(:backtrace)
         opts ||= {}
-        backtrace = caller_locations
-        while backtrace.first.path.end_with?("/logger.rb")
-          backtrace.shift
+        backtrace = message.backtrace if message.class <= ::Exception
+        backtrace ||= progname.backtrace if progname.class <= ::Exception
```

Comment:
I usually avoid operators when comparing classes, coz I never remember what they do specifically. Can this be done with e.g. `#kind_of?()` instead? 😃

Reply:
Huh interesting! I personally don't find it difficult to remember what they, but sure, I can go with `kind_of?`!